next-hardhat/test/index.ts

```diff
@@ -17,26 +16,40 @@ describe('It compiles noir program code, receiving circuit bytes and abi object.
 
   it('Should generate valid proof for correct input', async () => {
     const input = { x: 1, y: 2 };
-    await noir.init();
-    const witness = await noir.generateWitness(input);
-    correctProof = await noir.generateProof(witness);
 
-    expect(correctProof instanceof Uint8Array).to.be.true;
-  });
+    const backend = new BarretenbergBackend(circuit);
+    // Initialize program
+    const program = new Noir(circuit, backend);
+    // Generate proof
+    const proof = await program.generateFinalProof(input);
 
-  it('Should verify valid proof for correct input', async () => {
-    const verification = await noir.verifyProof(correctProof);
-    expect(verification).to.be.true;
+    // Proof verification
+    const isValid = await program.verifyFinalProof(proof);
+    expect(isValid).to.be.true;
   });
 
+  // TODO(Ze): My understanding is that this test was previously using a global noir instance
+  // TODO: Why was this the case and is this more indicative of the real workflow? 
+  // it('Should verify valid proof for correct input', async () => {
+  //   const verification = await program.verifyFinalProof(correctProof);
+  //   expect(verification).to.be.true;
+  // });
```

Comment:
TODO(Ze): not sure if we can just remove this

Reply:
I did it because that's how I usually do it. I think it's just a matter of style. I try to follow this pattern for tests:
- Use "describes" to separate common instances and keep them in that namespace (ex. `let noir : Noir;n`)
- Abuse "befores" to instantiate everything as close to the actual call being tested (ex. `noir = new Noir(backend, circuit)`)
- Test only the actual thing being tested (ex. `noir.generateProof()`)

Maybe it's just habit, as I've fought a lot with jest tests and nocks, mocks, spies, etc.

But I'm ok with bundling calls as you suggest 🙂